src/main/java/seedu/revised/command/subject/AccessSubjectCommand.java

```diff
@@ -40,7 +40,7 @@ public void execute(SubjectList subjectList, Storage storage)
         Subject gotoSubject = null;
 
         String[] message = this.fullcommand.split(" ");
-        if (message[1].isEmpty()) {
+        if (!fullcommand.contains(" ")) {
```

Comment:
What is this change for?

Reply:
split does not initialise message[1], so using the original method throws a IndexOutOfBoundsException while checking the if condition. The new method checks if there is only one word in fullcommand

Reply:
I see, it makes sense.

In this case if the users type in `subject     ` (extra spaces) I think it will go to the condition `if (gotoSubject == null)` and show 'cannot find the subject'

if we change it to `if (message.length < 2 || message[1].isEmpty())`, if will go in the condition and show NO_SUBJECT_EXCEPTION.

Do you think it would be better to show NO_SUBJECT_EXCEPTION or SUBJECT_NOT_FOUND_EXCEPTION in this case?

Reply:
Instead, we can trim the message so it will parse as if the user did not add extra spaces.